Reorder HPA resource list to work with GitOps tooling

Voziv opened the pull request:
<!--- Provide a general summary of your changes in the Title above --->
<!--- Please don't @-mention people in PR or commit messages (do so in an additional comment). --->

## What this PR does / why we need it:
<!--- Why is this change required? What problem does it solve? -->
<!--- If it fixes an open issue, please link to the issue here. -->
Using GitOps with ArgoCD currently causes infinite syncing when using both cpu and memory resource metrics for autoscaling.

Related: argoproj/argo-cd#1079
Related: kubernetes/kubernetes#74099

## Types of changes
<!--- What types of changes does your code introduce? Put an `x` in all the boxes that apply: -->
- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Which issue/s this PR fixes
<!--
(optional, in `fixes #<issue number>` format, will close that issue when PR gets merged):

fixes #
-->

## How Has This Been Tested?
<!--- Please describe in detail how you tested your changes. -->
<!--- Include details of your testing environment, and the tests you ran to -->
<!--- see how your change affects other areas of the code, etc. -->

Have had to make the same change in other places for the same reason. Kubernetes reorders the list order on the HPA to be Memory, then CPU.

This same change was applied to nginx-ingress in helm/charts#22626

## Checklist:
<!--- Go over all the following points, and put an `x` in all the boxes that apply. -->
<!--- If you're unsure about any of these, don't hesitate to ask. We're here to help! -->
- [x] My change requires a change to the documentation.
- [x] I have updated the documentation accordingly.
- [x] I've read the [CONTRIBUTION](https://github.com/kubernetes/ingress-nginx/blob/master/CONTRIBUTING.md) guide
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.

aledbf commented:
@Voziv please update the changelog and list mentioned in the comment 
https://github.com/kubernetes/ingress-nginx/blob/2d7b971358e9786cb8e6638749a7a5424b4da5e7/charts/ingress-nginx/Chart.yaml#L3-L4

aledbf commented:
/ok-to-test

Voziv commented:
Currently just editing via the github interface, will clone and squash the commits tomorrow.

@aledbf I've updated the change log, however I'm not sure how to edit the `artifacthub.io/changes` that the comment is referring to.

aledbf commented:
> Currently just editing via the github interface, will clone and squash the commits tomorrow.
> 
> @aledbf I've updated the change log, however I'm not sure how to edit the `artifacthub.io/changes` that the comment is referring to.

Just replace https://github.com/kubernetes/ingress-nginx/blob/master/charts/ingress-nginx/Chart.yaml#L23

Voziv commented:
Oh that's embarassing, I was totally looking at the site instead of in the file itself.

Updated and rebased.

I've also tested the change on a dev cluster and argocd no longer goes into an infinite loop, hurray!

aledbf commented:
/approve
/lgtm

aledbf commented:
@Voziv thanks!
